app/controllers/bookmarks_controller.rb

```diff
@@ -132,23 +132,27 @@ def status_action documents
     status = params['action']
     Array(documents.map(&:id)).each do |id|
       media_object = MediaObject.find(id)
-      if cannot? :update, media_object
-        errors += ["#{media_object.title} (#{id}) #{t('blacklight.messages.permission_denied')}."]
+      sucess = false
+      case status
```

Comment:
Then i would say break those out into an If elsif else like so: 

```ruby
if status.publish 
  success_ids << id if can?(:update, media_object)
elsif status.unpublish
  success_ids << id if can?(:unpublish, media_object)
else
  errors << #{media_object.title} (#{id}) #{t('blacklight.messages.permission_denied')}."
end
```

Regardless this is much simplier. Don't force a switch statement when you don't need it.